### ISSUE TITLE
Alternative solutions

### ISSUE THREAD

**Issue by relatedcode** [jessesquires/JSQMessagesViewController#2159]
I have been working with JSQMessagesViewController more than 3 years now. It is an amazing library. I love it so much. Thank you Jesse! :)

However it's no longer officially supported or maintained. So most of us needs to find an alternative chat user interface. Please help the community to find a great chat UI.

**Comment by 0xPr0xy** [comment]
@jessesquires Can you summarize the status of the SwiftExample in this repository compared to the JSQMessagesViewController and the project that you also worked on: https://github.com/MessageKit/MessageKit-iOS
What would you suggest for the community to move forward to?

**Comment by SD10** [comment]
I would be interested in organizing an effort to port this to Swift at the MessageKit repo and building an inclusive community. 

I think the open source iOS community benefits from having a centralized MessageViewController project as opposed to building competing, alternate solutions.  

There's no reason why there shouldn't be enough external support to continue moving this project forward.

**Comment by 0xPr0xy** [comment]
@SD10 https://github.com/relatedcode/ChatUIAlternatives/issues/1

**Comment by jessesquires** [comment]
@SD10 I completely agree.

Sadly, I also decided to remove `MessageKit`. It was a noble goal, but ultimately I cannot devote any time to it. I would *highly* encourage someone in the community to create a new `MessageKit` (or similar) organization and start building something new. 😄 A GitHub organization has many benefits over a project that is under one person's username.

I'm happy to provide advice, help, guidance where possible.

**Comment by 0xPr0xy** [comment]
hmm @relatedcode is just plugging his own `RCMessagesViewController` not even listing JSQMessagesViewController

**Comment by CavalcanteLeo** [comment]
https://github.com/mukyasa/MMTextureChat 

with texture (AsyncDisplayKit)

**Comment by SD10** [comment]
I took the initiative to replace the [MessageKit](https://github.com/SD10/MessageKit/issues/1) repo. 
It's currently hosted personally as I was unable to reserve the MessageKit organization name. 

I've opened an [issue](https://github.com/SD10/MessageKit/issues/1) to discuss the future of JSQMessagesViewController with anyone who shares my vision and would like to ensure that this project continues to move forward. I encourage anyone who wants to get involved to drop your $0.02 cents. Don't be shy 🤗 

@jessesquires Any guidance will always be greatly appreciated. I'm sure anyone looking to continue this project would listen to anything you have to say with open ears. While you may not have the resources to continue shipping code, your advice and experience around this topic are priceless.

**Comment by jyounus** [comment]
I really like @SD10's comment about building 1 ultimate Messaging lib for iOS, rather than having 100s of smaller, half-assed, competing solutions. I went ahead and secured the MessageKit org (really love the name), so if there's an interest from the community in building the ultimate Messaging UI Kit, we can use that. 

@SD10 I've got the org registered. I can go ahead and add you as an admin/transfer it to you if you want?

**Comment by SD10** [comment]
@jyounus Thank you for taking the initiative on something like that 👍 

Personally, I was able to reserve the `MessageKit` pod name. I truly love that name as well and was excited when I got a glimpse of Jesse's future vision for the project.

I think it's too early to transfer ownership but I would any support you can offer moving forward. 

Feel free to discuss anything in the other repo as I don't want to spam this issue.

**Comment by esilverberg** [comment]
I Googled "JSQMessagesViewController vs" and found that [SlackTextViewController](https://github.com/slackhq/SlackTextViewController) was the top hit. Can anyone with more experience comment on how equivalent the two projects are?

**Comment by pampas93** [comment]
Is there any way for me to get the previous framework of JSQMessageViewController? I use the pod now and am not able to use in my project. Please help.

**Comment by jessesquires** [comment]
@esilverberg - `SlackTextViewController` is only a text input bar component.

**Comment by jessesquires** [comment]
cc @MacMeDan and @Lucashuang0802, who have helped with JSQMVC and could be valuable members of this new org

**Comment by eliburke** [comment]
Derp. Sorry wrong button there. Was just trying to say: 
@jessesquires "SlackTextViewController is only a text input bar component." -- that's not precisely true. It doesn't provide pre-baked left/right cells or bubble images, but is otherwise a full messaging component.

**Comment by relatedcode** [comment]
I am just wondering if the MessageKit project will be continued here:

https://github.com/MessageKit

or here:

https://github.com/SD10/MessageKit

Thank! :)

**Comment by SD10** [comment]
@relatedcode The goal is to move the repo at SD10/MessageKit to MessageKit/MessageKit.

**Comment by relatedcode** [comment]
@SD10 Ok. Thanks! Please let me know once we can join to the final repo. :)

**Comment by jessesquires** [comment]
@eliburke -- oops! 😳 my bad

**Comment by jessesquires** [comment]
Everyone -- looks like we can move this conversation here:
https://github.com/MessageKit/MessageKit/issues/1

Going to close this now. 😊

**Comment by Juice007** [comment]
Thanks Jesse!

**Comment by umolac** [comment]
Sad to see such wonderful project retired. It was nice implementing and modifying it, although I actually never contributed here. It was, by my opinion, one of "must have" libraries if you're iOS dev. Thanks for wonderful time and for such wonderful project.

**Comment by acoroian** [comment]
this is what i started using some time ago. https://github.com/eBay/NMessenger

**Comment by iPrimeSystemSolution** [comment]
Hi everyone, I also want to contribute. I'm living in Brazil, but I want to help keep and continue this project. if you can call me privately, or send an email to me, I'll be grapto.developer@iprimebr.* com.br the asterisks are left over, then remove it.

**Comment by MacMeDan** [comment]
@iPrimeSystemSolution head on over to https://github.com/MessageKit/MessageKit thats where further development has commenced. 🤘🏼
